Standardize text prompting to connect/authenticate the user to wordpress.com

dereksmart opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->
Fixes inconsistent messaging to authenticate the user with WordPress.com

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
Updates all copy to use the same messaging for connecting/authenticating the user with WordPress.com.

Before - Dashboard: 
![image](https://user-images.githubusercontent.com/7129409/116405040-37d1e880-a7fd-11eb-8e93-e0f66d3d085c.png)

After - Dashboard: 
![image](https://user-images.githubusercontent.com/7129409/116412167-6901e700-a804-11eb-9f37-6dcd61d02776.png)

Before - Stats dashboard button:
![image](https://user-images.githubusercontent.com/7129409/116405194-5e901f00-a7fd-11eb-8887-ebfe7af1b797.png)

After - Stats dashboard button:
![image](https://user-images.githubusercontent.com/7129409/116405279-7667a300-a7fd-11eb-8477-bdf8cd23d762.png)

Before - Settings card bar:
(This is just one - there are many settings bars that change in the PR, please look at the code changes)
![image](https://user-images.githubusercontent.com/7129409/116405342-8a130980-a7fd-11eb-9cbe-f1f522ffe32f.png)

After - Settings card bar: 
![image](https://user-images.githubusercontent.com/7129409/116408347-ad8b8380-a800-11eb-8a13-8ae9d253be15.png)

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
p1HpG7-bRQ-p2#comment-46172

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Set up Jetpack in site-only connection mode, and view the respective areas. Check for:
- Spacing
- Typos
- Wording

folletto commented:
Since we are here, can we do these tweaks?

**Account connection**
Remove the duplicate, changing to a single sentence with the first part linked.

> Connect your WordPress.com account to get the most out of Jetpack

**Settings card bar**
Change the button to not wrap text, this could be as simple as "white-space: nowrap;".

As for the text, I'd tweak the left side:

> This feature is provided by the WordPress.com cloud. Connect to set up your status alerts.

Given the action is already the button, no need to repeat.

keoshi commented on projects/plugins/jetpack/_inc/client/sharing/publicize.jsx:
```diff
@@ -109,7 +109,10 @@ export const Publicize = withModuleSettingsFormHelpers(
 							featureLabel={ __( 'Publicize', 'jetpack' ) }
-							text={ __( 'Sign in to connect your social media accounts.', 'jetpack' ) }
+							text={ __(
+								'Connect your WordPress.com account to connect your social media accounts.',
```
This is the only one that raises some concerns to me — would suggested either of these with a slight preference for the first:

- Connect your WordPress.com account to manage your social media accounts.
- Connect your WordPress.com account to add your social media accounts.

dereksmart replied:
Done in 86f848d

keoshi reviewed (changes requested):
Thanks for doing this so quickly, @dereksmart ! This is _churrific_.

Only had one comment, everything else looks great.

dereksmart commented:
@keoshi @folletto great suggestions, thank you. I've implemented all suggestions and updated the screenshots in the OP. Settings cards look much better! 

I've taken this opportunity to match the disconnect user text (only shown to secondary connections) to match the connection text. See that change in ae9fcd3

Erin - the Account Connection card is a bit tricky - that link text is technically a button component, and would take a little bit of refactoring that I don't have time for right now. In the meantime, what do you think of this? 

![image](https://user-images.githubusercontent.com/7129409/116409796-2f2fe100-a802-11eb-83bd-3ac611294482.png)

folletto commented:
> Erin - the Account Connection card is a bit tricky - that link text is technically a button component, and would take a little bit of refactoring that I don't have time for right now. In the meantime, what do you think of this?

Ah! Thanks for the clarification there.

I think as a quick fix that works, thank you!

jeherve reviewed (approved):
Looks good to me.

jeherve commented:
Cherry-picked to `jetpack/branch-9.7` in a9cb8b53030783a0bf3bda04625433620f3121dc
